Intersections_2/include/CGAL/Intersection_traits_2.h

```diff
@@ -105,6 +105,28 @@ template<typename K>
 struct Intersection_traits<K, typename K::Triangle_2, typename K::Iso_rectangle_2>
   : public Intersection_traits<K, typename K::Iso_rectangle_2, typename K::Triangle_2> {};
 
+template<typename K, class B>
+struct Intersection_traits<K, CGAL::Bbox_2, B>  {
+  typedef typename Intersection_traits<K,typename K::Iso_rectangle_2, B>::result_type result_type;
+};
+
+template<typename K, class B>
+struct Intersection_traits<K, B, CGAL::Bbox_2>  {
+  typedef typename Intersection_traits<K,typename K::Iso_rectangle_2, B>::result_type result_type;
+};
+
+template<typename K>
+struct Intersection_traits<K, CGAL::Bbox_2, typename K::Point_2>  {
+  typedef typename boost::variant<typename K::Point_2> variant_type;
+  typedef boost::optional<variant_type> result_type;
+};
+
+template<typename K>
+struct Intersection_traits<K, typename K::Point_2, CGAL::Bbox_2>  {
+  typedef typename boost::variant<typename K::Point_2> variant_type;
+  typedef boost::optional<variant_type> result_type;
+};
```

Comment:
Because there are already partial specializations about `Point_2`:
https://github.com/maxGimeno/cgal/blob/c368b349d516e5e4649febd931658f0350344c66/Intersections_2/include/CGAL/Intersection_traits_2.h#L76-L93

Without those specific partial specializations, there would be an ambiguity between:
```c++
template<typename K, typename A>
struct Intersection_traits<K, A, typename K::Point_2>;
```
and 
```c++
template<typename K, class B>
struct Intersection_traits<K, CGAL::Bbox_2, B>;
```

Reply:
No idea :/